public/app/core/components/TimelineChart/utils.ts

```diff
@@ -405,22 +400,13 @@ export function prepareTimelineFields(
               },
             },
           };
-          changed = true;
-          fields.push(field);
+          frames.push({
+            fields: [frame.fields[startFieldIdx], f],
+            length: frame.length,
+          });
           break;
         default:
-          changed = true;
-      }
-    }
-    if (isTimeseries && fields.length > 1) {
```

Comment:
This is related to another comment I left, and something I am quite confused by:

In the old code, I think it's plausible to have, say, 3 time fields (no value field, e.g. number/string), and the logic will happily push a frame into `frames`. Is this an intentional feature?

Reply:
there can be one or two time fields (start times and end times). but we need a value field as well, obviously :)

Reply:
Yeah that part makes sense. I was more curious whether we should make this function explicitly handle/assert that there can be one or two, not >2. Anyhow I don't think this is relevant for this PR anymore, since I've reverted the `prepareTimelineFields` function. See a related comment https://github.com/grafana/grafana/pull/89586#discussion_r1660060463